treeherder/intermittents_commenter/commenter.py

```diff
@@ -0,0 +1,285 @@
+                change_priority, whiteboard = self.check_needswork_owner(change_priority, bug_info, whiteboard)
+
+            text = template.render(bug_id=bug_id,
+                                   total=counts['total'],
+                                   test_run_count=test_run_count,
+                                   rank=rank,
+                                   priority=priority,
+                                   failure_rate=round(counts['total']/float(test_run_count), 3),
+                                   repositories=counts['per_repository'],
+                                   platforms=counts['per_platform'],
+                                   startday=startday,
+                                   endday=endday.split()[0],
+                                   weekly_mode=self.weekly_mode)
+
+            params = {'bug_id': bug_id, 'comment': {'body': text}}
+            if whiteboard:
+                params['whiteboard'] = whiteboard
+            if change_priority:
+                params['priority'] = change_priority
+
+            all_params.append(params)
+
+        return all_params
+
+    def check_needswork_owner(self, change_priority, bug_info, whiteboard):
+        if (([bug_info['product'], bug_info['component']] in COMPONENTS) and
+            not self.check_whiteboard_status(whiteboard)):
+
+            if bug_info['priority'] not in ['--', 'P1', 'P2', 'P3']:
+                change_priority = '--'
+
+            stockwell_text = re.search(r'\[stockwell (.+?)\]', whiteboard)
+            if stockwell_text is not None and stockwell_text.group() != WHITEBOARD_NEEDSWORK_OWNER:
+                whiteboard = self.update_whiteboard(whiteboard, WHITEBOARD_NEEDSWORK_OWNER)
+
+        return change_priority, whiteboard
+
+    def check_needswork(self, whiteboard):
+        if whiteboard is not None:
+            stockwell_text = re.search(r'\[stockwell (.+?)\]', whiteboard)
+            # covers all 'needswork' possibilities, ie 'needswork:owner'
+            if stockwell_text is not None and stockwell_text.group(1).split(':')[0] == 'needswork':
+                whiteboard = self.update_whiteboard(whiteboard, WHITEBOARD_UNKNOWN)
+
+        return whiteboard
+
+    def assign_priority(self, priority, counts):
+        if priority == 0 and counts['total'] >= PRIORITY1_THRESHOLD:
+            priority = 1
+        elif priority == 0 and counts['total'] >= PRIORITY2_THRESHOLD:
+            priority = 2
+
+        return priority
+
+    def check_bug_info(self, bug_info, bug_id):
+        """check for previously fetched bug metadata"""
+        if not bug_info:
+            bug_info = self.fetch_bug_details(TRIAGE_PARAMS, bug_id)
+            if bug_info is None:
+                return None, None
+
+        return bug_info, bug_info['whiteboard']
+
+    def print_or_submit_comments(self, all_params):
+        for params in all_params:
+            if self.dry_run:
+                print(params['comment']['body'] + '\n')
```

Comment:
Could you switch this `print` to `logger.info` please?

Reply:
So you want this to log to a file rather rather than printing to console for testing locally?

Reply:
No, it will still go to `stdout` - using the Python/Django logging system is just best practice over using `print`. See:
https://docs.djangoproject.com/en/1.11/topics/logging/
https://github.com/mozilla/treeherder/blob/19daa928ed90e54ac17499eaa394a4102819576d/treeherder/config/settings.py#L207

Reply:
Ah, I was looking at Python's info on logging where it said that it doesn't print to console :)